### PR TITLE
[Helm] Parameterize resources for controller

### PR DESCRIPTION
**Pull request by craig-seeman**
## What does this PR do?

- [ ] Adds new functionality
- [X] Alters existing functionality
- [ ] Fixes a bug
- [ ] Improves documentation or testing

Please briefly describe your changes as well as the motivation behind them:
- `x` Sets resource allocation via helm variables instead of being hardcoded in the chart template.

## Code Quality Checklist

- [ ] The documentation is up to date.
- [ ] My code is sufficiently commented and passes continuous integration checks.
- [ ] I have signed my commit (see [Contributing Docs](../CONTRIBUTING.md)).

## Testing

- [ ] I leveraged continuous integration testing
    - [ ] by depending on existing `unit` tests or `end-to-end` tests.
    - [ ] by adding new `unit` tests or `end-to-end` tests.
- [ ] I manually tested the following steps:
    - `x`
    - [ ] locally.
    - [ ] as a canary deployment to a cluster.

### CODE REVIEW

**Review by ptnapoleon — APPROVED** [review]

**Review by ptnapoleon — COMMENTED** [review]
We don't use these limit values in production, just for dev, we set them according to our typical usage in real clusters. It would be better to move these to `values.yaml` and configure them however you'd like, without needing to change `deployment.yaml` in the future

**Comment by craig-seeman** [comment]
> We don't use these limit values in production, just for dev, we set them according to our typical usage in real clusters. It would be better to move these to `values.yaml` and configure them however you'd like, without needing to change `deployment.yaml` in the future

I like this approach! I'll shoot that into a lil' refactor tomorrow here and do that~ Thanks for the feedback!

**Comment by Devatoria** [comment]
Thanks for the contribution. We also have people here working on improving the overall memory usage of the controller to make it more stable because it can spike for large experiments indeed. We used to have to temporarily increase that limit during some injections on a few thousands targets at Datadog otherwise it would OOM too. Hopefully we have a bunch of fixes soon to help with that.

**Comment by craig-seeman** [comment]
Any strong feelings on the level of implementation? 

After looking at the config with the resources defined, I'm sort of leaning more towards a simpler config file (A) and just specifying the limit and request in the same setting as opposed to allow both the request and limit to be set individually (B).

(A)
```yaml
resources:
  cpu: 100m
  memory: 700Mi
```

(B)
```yaml
resources:
  limits:
    cpu: 100m
    memory: 700Mi
  requests:
    cpu: 100m
    memory: 700Mi
```

**Comment by craig-seeman** [comment]
Went with the (A) Approach outlined above to keep simplicity around the configs. I also went ahead and set the default back to 300Mi after realizing _we_ might be the exception vs the norm in our deployment size.

File: chart/values.yaml [hunk 1]
```diff
@@ -57,6 +57,9 @@ controller:
     enable: false
     namespaceThreshold: 80
     clusterThreshold: 66
+  resources: # resources assinged to the controller pod. may need to be increased when deploying to larger scale clusters
```
Comment (ptnapoleon) [inline chart/values.yaml:60]:
```suggestion
  resources: # resources assigned to the controller pod. may need to be increased when deploying to larger scale clusters
```

**Comment by ptnapoleon** [comment]
One nit and then I'll merge. Definitely like the approach of keeping limits and requests identical for the controller

**Pull request merged**